raft-tech/TANF-app
# Feat/2369 tanf t1 validation

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2369 

## How to Test

1. Deploy locally or to dev
2. Go to datafile page and upload a file
3. Section 1 t1 data will be parsed and validated errors will be collected in a list of dictionaries
4. There is no response to the user yet, but the app should run without failing

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

**Acceptance Criteria:** 

- [x] Parsing completes regardless of data quality
- [x] Parsing errors whether fatal or warning generate an accessible data structure

**Tasks:**

- [x] Implement validator functions for both warning and fatal edits/errors including relevant unit tests.
- [x] Fields include item number and item name per docs
- [x] Extend RowSchema and Field classes to co-locate a list of validator functions within the Field under `search_indexes/parsers/scehma_defs/tanf.py`

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: 92.06% (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: 94.11% (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
